codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1277,10 +1278,27 @@ addToDependentsIndex dependency dependent = execute sql (dependency :. dependent
     ON CONFLICT DO NOTHING
   |]
 
--- | Get non-self, user-defined dependents of a dependency.
-getDependentsForDependency :: Reference.Reference -> Transaction [Reference.Id]
-getDependentsForDependency dependency =
-  filter isNotSelfReference <$> queryListRow sql dependency
+-- | Which dependents should be returned?
+--
+-- * /IncludeAllDependents/. Include all dependents, including references from one's own component-mates, and references
+-- from oneself (e.g. those in recursive functions)
+-- * /ExcludeSelf/. Include all dependents, including references from one's own component-mates, but excluding
+-- actual self references (e.g. those in recursive functions).
+-- * /ExcludeOwnComponent/. Include all dependents outside of one's own component.
+data DependentsSelector
```

Comment:
I definitely appreciate the note here explaining each of the selectors, but it's still maybe a bit strange to switch between "Include" and "Exclude" between them. For settings like this they're referenced very rarely so maybe we just make it a long name? Or optionally make it a product type with `(includeSelf, includeComponentMates, includeExternalDeps)`?

Reply:
Hmm, I'm not sure what to do about this suggestion. I think the naming is ok as-is. What names do you have in mind?

Reply:
Do we like `All | AllButComponent | AllButSelf`?  I agree the current names seem a little weird but I can't think of anything better, so leaving them as-is seems fine..

Reply:
Is the purpose of this renaming to make each data construcor begin with the same slug like `All`?

Otherwise, I think the names in the PR are equally or more clear -

`All` -> `IncludeAllDependents`
`AllButComponent` -> `ExcludeOwnComponent`
`AllButSelf` -> `ExcludeSelf`

Reply:
I personally prefer Arya's suggestion, but it's really just a nit, and the most important is that there are docs on the type.

Reply:
Ok, let's rename in another PR then, if we want to